release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md

```diff
@@ -36,11 +36,135 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
 
 ### New features
 
+##### Support to import a non-Ballerina interface definition as a module that defines a client
+
+A module client declaration can be used with a supported non-Ballerina interface definition to generate a client module corresponding to the definition. The prefix used in the client declaration can then be used to access constructs from the generated module.
+
+```ballerina
+import ballerina/io;
+
+client "petstore.yaml" as petstore;
+
+public function main() returns error? {
+    petstore:client petstoreClient = check new;
+
+    petstore:Pets pets = check petstoreClient->/pets();
+
+    foreach petstore:Pet pet in pets {
+        string? petType = pet.'type;
+
+        if petType is string {
+            io:println(string `Pet type: ${petType}, name: ${pet.name}`);
+        } else {
+            io:println(string `Pet name: ${pet.name}`);
+        }
+    }
+}
+```
+
+The reserved `client` keyword can be used in a qualified identifier (`petstore:client`) without a quoted identifier. This is specifically allowed when the module prefix (`petstore`) is defined using a client declaration.
+
+The Swan Lake Update 3 release supports OpenAPI and GraphQL definitions in client declarations.
+
+#### Added a new field to display the annotation
+
+A new metadata called `kind` has been introduced to `display` the annotation in order to categorize the kind of the record field.
+
+```ballerina
+public type RefreshTokenGrantConfig record {|
+    @display {iconPath: "Field.icon", label: "clientSecret field", kind: "password"}
+    string clientSecret;
+|};
+```
+
+#### Added support for function pointers with defaultable parameters
+
+Function pointers are now allowed with default values for function pointer parameters. Any expression can be used as the default value of a function pointer parameter.
+
+```ballerina
+public function main() {
+	int num1 = 100;
+	int num2 = 50;
+	int num3 = 25;
+
+	function(int a = 0, int b = 0, int c = 0) returns int total = getSum;
+    
+	io:println(total()); // Prints `0`.
+	io:println(total(num1, num2)); // Prints `150`.
+	io:println(total(num1, num2, num3)); // Prints `175`.
+}
+
+function getSum(int num1, int num2, int num3) returns int {
+	return num1 + num2 + num3;
+}
+```
+
 ### Improvements
 
-### Bug fixes
+#### More improvements on working with optional fields
+
+If there is an optional field `T x?;` in a record, the absence of `x` is represented by nil where `T` does not allow nil.
+
+```ballerina
+type Topt record {
+    int x?;
+    int y?;
+};
+
+public function main() {
+    Topt t = {x: 2, y: 4};
+    t.x = ();
+
+    t = {x: 2};
+    var {x: _, y} = t;
+    io:println(y is ()); // true
+
+    int? xOrNil = ();
+    t = {x: xOrNil, y: 5};
+    io:println(t.x is ()); // true
+}
+
```

Comment:
Shall we improve this example to use a real-world use case (e.g., employee, student, etc.)?